bitsandbytes/nn/modules.py

```diff
@@ -481,10 +481,8 @@ def forward(self, x: torch.Tensor):
             x = x.to(self.compute_dtype)
 
         bias = None if self.bias is None else self.bias.to(self.compute_dtype)
-        out = bnb.matmul_4bit(x, self.weight.t(), bias=bias, quant_state=self.weight.quant_state)
-
-        out = out.to(inp_dtype)
 
+        out = bnb.matmul_4bit(x, self.weight.t(), bias=bias, quant_state=self.weight.quant_state).to(inp_dtype)
         return out
```

Comment:
While you're at it:
```suggestion
        return bnb.matmul_4bit(x, self.weight.t(), bias=bias, quant_state=self.weight.quant_state).to(inp_dtype)
```

Reply:
There's a flake8 rule set available in Ruff (RET) that covers things like this; may separately enable that.

Reply:
Yep, but figured since you were already coalescing the `.to` above, might as well do the obvious next thing now.

Reply:
Updated!